patches/minecraft/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.java.patch

```diff
@@ -1,5 +1,29 @@
 --- a/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.java
 +++ b/net/minecraft/client/gui/screens/worldselection/CreateWorldScreen.java
+@@ -122,7 +_,10 @@
+          return new WorldCreationContext(p_232884_, Lifecycle.stable(), p_232883_, p_232882_);
+       }, Util.m_183991_(), p_232897_);
+       p_232897_.m_18701_(completablefuture::isDone);
+-      p_232897_.m_91152_(new CreateWorldScreen(p_232898_, DataPackConfig.f_45842_, new WorldGenSettingsComponent(completablefuture.join(), Optional.of(WorldPresets.f_226437_), OptionalLong.empty())));
++      // FORGE: Force load mods' datapacks after setting screen.
++      CreateWorldScreen createWorldScreen = new CreateWorldScreen(p_232898_, DataPackConfig.f_45842_, new WorldGenSettingsComponent(completablefuture.join(), Optional.of(WorldPresets.f_226437_), OptionalLong.empty()));
++      p_232897_.m_91152_(createWorldScreen);
++      createWorldScreen.m_100878_(packrepository);
+    }
+ 
+    public static CreateWorldScreen m_205426_(@Nullable Screen p_205427_, WorldStem p_205428_, @Nullable Path p_205429_) {
+@@ -436,7 +_,10 @@
+          return !list.contains(p_232927_);
+       }).collect(ImmutableList.toImmutableList());
+       DataPackConfig datapackconfig = new DataPackConfig(list, list1);
+-      if (list.equals(this.f_100846_.m_45850_())) {
++      // FORGE: Force load and validate worldgen data when creating new world or rearranging datapacks.
++      // Force load on new world is necessary to ensure mods' worldgen  is loaded.
++      // Force load when leaving datapack rearrange screen is necessary to ensure invalid datapacks from the other force load continue to be recognized as invalid.
```

Comment:
Can you explain a bit more why this force load when leaving the datapack rearrange screen is necessary? Don't really get it.

Reply:
So currently (without this fix), anything that any mod adds in data/worldgen is simply not loaded at all. This also includes the new BiomeModifiers. So mob spawns and similar does not work either in Forge currently. 
You can force it to load by re-arranging the datapacks in the datpack menu, when creating a new world.
Hope that simplifies it a bit :)

Reply:
I understand the issue and that this fixes it. I do not understand the comment in this patch.

Reply:
Better fix here would be to change the `DataPackConfig.DEFAULT` in `openFresh` to be `new DataPackConfig(ImmutableList.of("vanilla"), ImmutableList.of())`  which is what the completable future atually does.
The reason you have to noop this if is because DEFAULT has our mods in it. 
IF it is possible, killing the entire completable future thing in openFresh would be the best option as it's just a waste of time.

Reply:
Alright, wasn't sure if it would be a pain to do but it would be better.